scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -548,6 +646,72 @@ def add_testsuites(self, testsuite_filter=None):
                                 suite.platform_allow,
                                 f"platform_allow in {suite.name}")
 
+                        if self.options.check:
+                            checker = self.checker
+                            if len(suite.platform_allow) > MAX_PLATFORM_ALLOW:
+                                checker.add(
+                                    checker.ALLOW,
+                                    f"Too many entries in allow_platform in {suite.name}: "
+                                    f"{len(suite.platform_allow)}"
+                                )
+
+                            if len(suite.integration_platforms) > MAX_INTEGRATION_PLATFORMS:
+                                checker.add(
+                                    checker.INTEGRATION,
+                                    f"Too many integration platforms in {suite.name}: "
+                                    f"{len(suite.integration_platforms)}"
+                                )
+
+                            if len(suite.platform_exclude) > MAX_PLATFORM_EXCLUDE:
+                                checker.add(
+                                    checker.EXCLUDE,
+                                    f"Too many excluded platforms in {suite.name}: "
+                                    f"{len(suite.platform_exclude)}"
+                                )
+
+                            if len(suite.platform_allow) > 1 and not suite.integration_platforms:
+                                checker.add(
+                                    checker.ALLOW_NO_INTEGRATION,
+                                    f"platform_allow in {suite.name} without integration_platforms"
+                                )
```

Comment:
not sure how helpful is this check. Why not just checking if integration_platforms are provided or not (regardless of platform_allow)? Or from another perspective: when it is not an issue when there are no integration_platforms?